lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -46,27 +47,32 @@ def __init__(self, charm, substrate="k8s"):
         super().__init__(charm, "client-relations")
         self.charm = charm
         self.substrate = substrate
-        self.framework.observe(self.charm.on.config_changed, self._on_pbm_config_changed)
-
-    def _on_pbm_config_changed(self, event) -> None:
-        """Handles PBM configurations."""
-        # handling PBM configurations requires that the pbm snap is installed.
-        if "db_initialised" not in self.charm.app_peer_data:
-            logger.debug("Cannot set PBM configurations, MongoDB has not yet started.")
-            event.defer()
+        self.framework.observe(self.charm.on.create_backup_action, self._on_create_backup_action)
+
+    def _on_create_backup_action(self, event) -> None:
+        try:
+            snap_cache = snap.SnapCache()
+            pbm_snap = snap_cache["percona-backup-mongodb"]
+            if not pbm_snap.present:
+                logger.debug("Cannot start PBM agent, PBM snap is not yet installed.")
+                event.defer()
+                return
+            pbm_snap.start(services=["pbm-agent"])
```

Comment:
just Q: is it possible to configure storage before `pbm-agent` is started?

Reply:
Yes, it is :)